tests/build/test_generate_config.py

```diff
@@ -395,3 +395,35 @@ def test_settings_from_multiple_libs_included(matching_target_and_filter, progra
 
     assert "MBED_CONF_PLATFORM_STDIO_BAUD_RATE=9600" in config_text
     assert "MBED_LFS_READ_SIZE=64" in config_text
+
+
+def test_requires_config_option(program):
```

Comment:
Was this test failing before we added the `requires` handling? Do we also need a test to ensure we're filtering out things we don't `require`?

Reply:
Yes, this test will fail without `requires` handling.

This test combines both checking for config options that applications requires and don't require.
```
assert "MBED_CONF_PLATFORM_STDIO_BAUD_RATE=9600" in config_text
    assert "MBED_LFS_READ_SIZE=64" not in config_text
```
Should I split them to different tests? Most of the contents of the test will remain the same. So I decided to put them together.

Reply:
OK. I missed the `not` in the second assertion when I was looking over the code. Keeping it all in one test is fine.